### PR TITLE
rc6: RC6 Implementation

### PR DESCRIPTION
**Pull request by ashWhiteHat**
Part of https://github.com/RustCrypto/block-ciphers/issues/1

Hi there.
I implemented rc6.

Implementation and documentation are aligned with rc5 implementation.

I referred following document for key schedule, encryption and decryption algorithms.
https://www.grc.com/r&d/rc6.pdf

I checked whether it works by following test vectors.
https://datatracker.ietf.org/doc/html/draft-krovetz-rc6-rc5-vectors-00#section-3

I would appreciate it if you could confirm.
Thank you!

### CODE REVIEW

**Comment by NamedNeon** [comment]
+1

RustCrypto apparently already owns the crate "rc6" and just isn't doing anything with it, so not merging this seems somewhat silly.

**Comment by tarcieri** [comment]
The reason this hasn't been merged yet is because it hasn't yet been code reviewed, which is especially important for cryptographic code.

Please give us time to review it.

**Comment by NamedNeon** [comment]
> The reason this hasn't been merged yet is because it hasn't yet been code reviewed, which is especially important for cryptographic code.
> 
> Please give us time to review it.

Oh, yeah, 100%. I apologize if I came off as impatient. My intent was to express interest in the feature, not to rush you guys.  

> If you'd like to be helpful, it would be good to get some feedback on if it worked in a particular application you're interested in.

I am planning on using it in an application soon, hence my original comment. I'll report back with my results.

**Comment by NamedNeon** [comment]
The RC6 standard doesn't mandate any key size, round count, or word length, but this implementation does. RustCrypto's RC5 implementation has a similar problem which I've opened an issue for (#381). RC5 and RC6 are very similar so most of what I wrote there should be applicable here.

**Comment by tarcieri** [comment]
The test failures look unrelated.

We'd still be interested in this if you could rebase!

**Comment by ashWhiteHat** [comment]
Hi @tarcieri 
Thank you for the notification.

I rebased and fixed conflict.
I would appreciate it if you could confirm.
Thank you!

File: rc6/Cargo.toml [hunk 1]
```diff
@@ -0,0 +1,21 @@
...
+authors = ["RustCrypto Developers"]
+edition = "2021"
+license = "MIT OR Apache-2.0"
+readme = "README.md"
+repository = "https://github.com/RustCrypto/block-ciphers"
+keywords = ["crypto", "rc6", "block-cipher"]
+categories = ["cryptography"]
+
+[dependencies]
+cipher = { version = "0.4.3", features = ["zeroize"] }
```
Comment (tarcieri) [inline rc6/Cargo.toml:14]:
```suggestion
cipher = { version = "0.5.0-pre.6", features = ["zeroize"] }
```

File: rc6/Cargo.toml [hunk 2]
```diff
@@ -0,0 +1,21 @@
...
+
+[dev-dependencies]
+cipher = { version = "0.4.3", features = ["dev"] }
```
Comment (tarcieri) [inline rc6/Cargo.toml:17]:
```suggestion
cipher = { version = "0.5.0-pre.6", features = ["dev"] }
```

**Comment by ashWhiteHat** [comment]
This includes rc5 refactoring and it's hard to rebase.
I close this PR.

**Comment by tarcieri** [comment]
@ashWhiteHat can you open a new PR with just the `rc6` implementation?

**Comment by ashWhiteHat** [comment]
Copy that.
Probably in weekend.

**Pull request closed without merging**